stylix/testbed/default.nix

```
@@ -6,26 +6,35 @@
 }:
 
 let
+  availableGraphicalEnvironments = lib.pipe ./graphicalEnvironments [
+    builtins.readDir
+    (lib.mapAttrsToList (name: _: lib.removeSuffix ".nix" name))
+  ];
+
   makeTestbed =
     name: testbed:
     let
 
       system = lib.nixosSystem {
         inherit (pkgs) system;
 
-        modules = [
-          ./modules/common.nix
-          ./modules/enable.nix
-          ./modules/application.nix
-          inputs.self.nixosModules.stylix
-          inputs.home-manager.nixosModules.home-manager
-          testbed
+        modules =
+          [
+            ./modules/common.nix
+            ./modules/enable.nix
+            ./modules/application.nix
+            inputs.self.nixosModules.stylix
+            inputs.home-manager.nixosModules.home-manager
+            testbed
 
-          # modules for external targets
-          inputs.nvf.nixosModules.default
-          inputs.nixvim.nixosModules.nixvim
-          inputs.spicetify-nix.nixosModules.spicetify
-        ];
+            # modules for external targets
+            inputs.nvf.nixosModules.default
+            inputs.nixvim.nixosModules.nixvim
+            inputs.spicetify-nix.nixosModules.spicetify
+          ]
+          ++ map (
+            name: import ./graphicalEnvironments/${name}.nix
+          ) availableGraphicalEnvironments;
```

What I meant with https://github.com/nix-community/stylix/pull/1222#discussion_r2202860984 and https://github.com/nix-community/stylix/pull/1222#discussion_r2202863005 is to inline the `availableGraphicalEnvironments` identifier. Not to [remove and inline the called `available-graphical-environments.nix` file](https://github.com/nix-community/stylix/compare/5a1eebf1bae4be94b92cce7adbf184bae5113a5b..0c5c1f76369dde10c3b10b08621847c69347aad3). I would have been fine with duplicating this small amount of code, but since you already proposed it, you might as well keep it.

```suggestion
          ) (import ./available-graphical-environments.nix { inherit lib; });
```

---

Oh... :p